### PR TITLE
errbot-reminder: send notifications before meetings

### PR DESCRIPTION
**Pull request by adrien5054**
* Send notifications before daily meetings

### CODE REVIEW

**Comment by adrien5054** [comment]
On the last commit, on lines 34 and 38, I forgot the '()' after 'today'

File: errbot-reminder.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+from errbot import BotPlugin, botcmd
+from datetime import datetime, time, timedelta, date
+
+# Daily / Retrospective --> If it's an ordinary day or a retrospective, meeting at 9:30
+# Sprint Planning --> If it's a Sprint planning, meeting at 15:30
+# Sprint review --> If it's a Sprint Review, meeting at 14:45
+
+class PluginTemplate(BotPlugin):
```
Comment (benjamin-bergia) [inline errbot-reminder.py]: should be your plugin name so:
```suggestion
class Reminder(BotPlugin):
```

File: errbot-reminder.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  @botcmd
+  def whereami(self, msg, args):
```
Comment (benjamin-bergia) [inline errbot-reminder.py]: This is the name of the command you will run in the chat. 
Unfortunately you can't use a command for this. We want this to run automatically at regular intervals.

**Review by benjamin-bergia — CHANGES_REQUESTED** [review]

File: errbot-reminder.py (hunk 1 above)
Comment (adrien5054) [reply errbot-reminder.py]: It's done

File: errbot-reminder.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+# Sprint review --> If it's a Sprint Review, meeting at 14:45
+#  -----------------
+
+
+class Reminder(BotPlugin):
+    @botcmd
+    def whereami(self, msg, args):
+        stream = msg._from._room._id
+        topic = msg._from._room._subject
+        # return f"You are in #{stream}>{topic}"
```
Comment (drjova) [inline errbot-reminder.py]: could you please remove the comments inside the class?

Comment (adrien5054) [reply errbot-reminder.py]: It's done

**Comment by drjova** [comment]
@adrien5054 please squash the commits

**Comment by adrien5054** [comment]
> @adrien5054 please squash the commits

it's ok

**Review by drjova — APPROVED** [review]

File: errbot-reminder.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        day_of_the_week = datetime.now().weekday()
+        current_time = datetime.now().replace(second=0, microsecond=0)
+        stream = "tools & services"
+
+        if day_of_the_week != 5 and day_of_the_week != 6:
+            if date.today().weekday() == 0 and is_sprint_planning:
+                meeting_time = time(15, 30)
+                topic = "sprint planning"
+
+            elif date.today().weekday() == 3 and not is_sprint_planning:
```
Comment (drjova) [inline errbot-reminder.py]: the `is_sprint_planning` is never called, it is a function

File: errbot-reminder.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            monday = datetime.combine(monday, datetime.min.time())
+            return monday
+
+        def is_sprint_planning():
+            nb_days = first_iteration_startdate - get_monday()
+            nb_weeks = nb_days / 7
+            nb_weeks = int(nb_weeks.total_seconds())
+
+            if nb_weeks % 2 == 0:
+                return True
```
Comment (drjova) [inline errbot-reminder.py]: We don't need to have nested functions

File: errbot-reminder.py (hunk 4 above)
Comment (drjova) [reply errbot-reminder.py]: This applies to all the code

Comment (adrien5054) [reply errbot-reminder.py]: I call this function at the lines 40, 44 and 48. But I forgot to put the brackets, sorry

File: errbot-reminder.py [hunk 6]
```diff
@@ -10,42 +10,41 @@
...
-            monday = datetime.combine(monday, datetime.min.time())
-            return monday
+        if nb_weeks % 2 == 0:
+            return True
 
-        def is_sprint_planning():
-            nb_days = first_iteration_startdate - get_monday()
-            nb_weeks = nb_days / 7
-            nb_weeks = int(nb_weeks.total_seconds())
+    def whereami(self, msg, args):
```
Comment (drjova) [inline errbot-reminder.py]: `whereami` needs the decorator `botcmd`

Comment (adrien5054) [reply errbot-reminder.py]: I corrected it

File: errbot-reminder.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        first_iteration_startdate = datetime(2022, 2, 28)
+        nb_days = first_iteration_startdate - self.get_monday()
+        nb_weeks = nb_days / 7
+        nb_weeks = int(nb_weeks.total_seconds())
+
+        if nb_weeks % 2 == 0:
+            return True
+
+    @botcmd
+    def whereami(self, msg, args):
```
Comment (drjova) [inline errbot-reminder.py]:
```suggestion
    def notify_for_daily_meeting(self, msg, args):
```

**Review by drjova — APPROVED** [review]
@benjamin-bergia would you like to have a final look?

File: errbot-reminder.plug [hunk 8]
```diff
@@ -0,0 +1,10 @@
+[Core]
+name = errbot-reminder
```
Comment (benjamin-bergia) [inline errbot-reminder.plug]:
```suggestion
name = Reminder
```

File: errbot-reminder.plug [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+module = github
```
Comment (benjamin-bergia) [inline errbot-reminder.plug]:
```suggestion
module = reminder
```

File: errbot-reminder.plug (hunk 8 above)
Comment (benjamin-bergia) [reply errbot-reminder.plug]: This is the name of the plugin as displayed on Errbot

File: errbot-reminder.plug (hunk 9 above)
Comment (benjamin-bergia) [reply errbot-reminder.plug]: This is the name of your module/file.

File: errbot-reminder.py
Comment (benjamin-bergia) [inline errbot-reminder.py:1]: The standard we use is to name the module without `errbot-` so in this case `reminder.py`

File: errbot-reminder.py [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+        return monday
+
+    def is_sprint_planning(self):
+        first_iteration_startdate = datetime(2022, 2, 28)
+        nb_days = first_iteration_startdate - self.get_monday()
+        nb_weeks = nb_days / 7
+        nb_weeks = int(nb_weeks.total_seconds())
+
+        if nb_weeks % 2 == 0:
+            return True
```
Comment (benjamin-bergia) [inline errbot-reminder.py]:
```suggestion
        return nb_weeks % 2 == 0
```

**Review by benjamin-bergia — CHANGES_REQUESTED** [review]

**Review by benjamin-bergia — APPROVED** [review]

**Pull request merged**